docs/website/docs/building-from-source/getting-started.md

```diff
@@ -70,66 +67,75 @@ cd iree
 git submodule update --init
 ```
 
-Configure then build all targets using CMake:
+The most basic CMake workflow is:
+
+``` shell
+# Configure
+cmake -G Ninja -B ../iree-build/ .
+
+# Build
+cmake --build ../iree-build/
+```
+
+!!! Caution "Caution - slow builds"
+    The compiler build is complex. You will want a powerful machine and to tune
+    the settings following the next section.
+
+    Use case permitting, disabling the compiler build with
+    `-DIREE_BUILD_COMPILER=OFF` will drastically simplify the build.
+
+## Configuration settings
 
-Configure CMake:
+The configure step should be customized for your build environment. These
+settings can improve compile and link times substantially.
+
+<!-- TODO(scotttodd): add notes about CMake presets?  -->
 
 === "Linux"
 
     ``` shell
-    # Recommended for simple development using clang and lld:
-    cmake -GNinja -B ../iree-build/ -S . \
+    # Recommended development options using clang and lld:
+    cmake -G Ninja -B ../iree-build/ -S . \
```

Comment:
If recommending a debug build, we really should be directing Linux users to include:

```
-DIREE_ENABLE_SPLIT_DWARF=ON
-DIREE_ENABLE_THIN_ARCHIVES=ON
```

Especially for more limited machines, these have massive space/overhead savings when debug symbols are present.

Reply:
They really only should be for "dev" use as they make the artifacts non-relocatable and such.